commands/create.go

```diff
@@ -61,32 +61,19 @@ func runCreate(dockerCli command.Cli, in createOptions, args []string) error {
 		}
 	}
 
-	buildkitHost := os.Getenv("BUILDKIT_HOST")
-
-	driverName := in.driver
-	if driverName == "" {
-		if len(args) == 0 && buildkitHost != "" {
-			driverName = "remote"
-		} else {
-			var arg string
-			if len(args) > 0 {
-				arg = args[0]
-			}
-			f, err := driver.GetDefaultFactory(ctx, arg, dockerCli.Client(), true)
-			if err != nil {
-				return err
-			}
-			if f == nil {
-				return errors.Errorf("no valid drivers found")
+	if !in.actionLeave && !in.actionAppend {
+		contexts, err := dockerCli.ContextStore().List()
+		if err != nil {
+			return err
+		}
+		for _, c := range contexts {
+			if c.Name == in.name {
+				logrus.Warnf("instance name %q already exists as context builder", in.name)
+				break
```

Comment:
Any reason to move that logic before https://github.com/docker/buildx/blob/3cf549a7f7ee015cabe99d4a08e4b9aabfc83ccc/commands/create.go#L96-L102

If `in.name` is empty it could still be the same name as an existing context when generated.